### PR TITLE
ADS: Add Checkbox to TextAlertDialog

### PR DESCRIPTION
**Pull request by malmstein**
Task/Issue URL: https://app.asana.com/0/1174433894299346/1208611782328138/f

### Description
This PR updates the TextAlertDialog with a few things:
- Adds a checkbox
- Allows for button configuration

The destructive combination was using the wrong type of buttons, so we added a GhostAlt version that can now be used.

### Steps to test this PR

_ADS Demo_
- [x] Open ADS Demo
- [x] Navigate to Dialogs tab
- [ ] Verify dialogs look as expectec\

### CODE REVIEW

**Comment by malmstein** [comment]
* **#5192** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5192?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @malmstein and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by nalcalag — CHANGES_REQUESTED** [review]
@malmstein `Single Choice Destructive Alert Dialog` doesn't look as expected. Buttons margin are wrong:
Screenshot_20241025_152055_DuckDuckGo

**Comment by malmstein** [comment]
@nalcalag this issue happens in Prod, it’s not related to this PR. That being said, I’m going to fix it because it shouldn’t be like that.

File: app/src/main/java/com/duckduckgo/app/launch/LaunchBridgeActivity.kt [hunk 1]
```diff
@@ -39,6 +40,8 @@ class LaunchBridgeActivity : DuckDuckGoActivity() {
 
         setContentView(R.layout.activity_launch)
 
+        Timber.d("pixel request: launch")
+
```
Comment (nalcalag) [inline app/src/main/java/com/duckduckgo/app/launch/LaunchBridgeActivity.kt:44]: I guess this is a log added by mistake?

File: app/src/main/res/layout/view_new_omnibar_bottom.xml [hunk 2]
```diff
@@ -26,6 +26,7 @@
 
     <androidx.constraintlayout.widget.ConstraintLayout
         android:id="@+id/toolbarContainer"
+        android:elevation="2dp"
```
Comment (nalcalag) [inline app/src/main/res/layout/view_new_omnibar_bottom.xml:29]:
```suggestion
```

File: common/common-ui/src/main/res/layout/dialog_text_alert.xml [hunk 3]
```diff
@@ -52,50 +54,28 @@
...
-        app:layout_constraintEnd_toEndOf="parent"
-        app:layout_constraintTop_toBottomOf="@+id/textAlertDialogMessage"/>
+        app:typography="body1" />
 
-    <com.duckduckgo.common.ui.view.button.DaxButtonDestructive
-        android:id="@+id/textAlertDialogPositiveDestructiveButton"
-        android:layout_width="wrap_content"
+    <CheckBox
+        android:id="@+id/textAlertDialogCheckBox"
+        android:layout_width="0dp"
```
Comment (nalcalag) [inline common/common-ui/src/main/res/layout/dialog_text_alert.xml:61]: _Question_: why `0dp` instead `wrap_content` if we are not setting the end constraint?

**Review by nalcalag — APPROVED** [review]
LGTM! Plenty of requests from ADS done in the PR 🏅 
I think there are a couple of lines added by mistake from other PRs and a question, but looks great!

File: app/src/main/java/com/duckduckgo/app/launch/LaunchBridgeActivity.kt (hunk 1 above)
Comment (malmstein) [reply app/src/main/java/com/duckduckgo/app/launch/LaunchBridgeActivity.kt:44]: Yes, removing!

File: common/common-ui/src/main/res/layout/dialog_text_alert.xml (hunk 3 above)
Comment (malmstein) [reply common/common-ui/src/main/res/layout/dialog_text_alert.xml:61]: No reason, will revert because that makes sense.

**Comment by malmstein** [comment]
Thanks for the review @nalcalag !

**Pull request merged**